wrapanapi/systems/lenovo.py

```diff
@@ -453,3 +479,121 @@ def get_device_unique_id(device):
+        return self.raw.get("partNumber", None)
+
+    @property
+    def description(self):
+        return self.raw.get("description", None)
+
+    @property
+    def firmwares(self):
+        return self.raw.get('firmware', None)
+
+    @property
+    def power_status(self):
+        return self.raw.get('powerState', None).lower()
+
+    @property
+    def type(self):
+        return self.raw.get('type', None)
+
+    @property
+    def health_state(self):
+        return self.raw.get('overallHealthState', None)
+
+    @property
+    def ipv4_addresses(self):
+        return self.raw.get('ipv4Addresses', None)
+
+    @property
+    def ipv6_addresses(self):
+        return self.raw.get('ipv6Addresses', None)
+
+    @property
+    def _identifying_attrs(self):
+        return {"name": self.name, "uuid": self.uuid}
+
+    def refresh(self):
+        uri = "{}/{}".format(LenovoSwitch.API_PATH, self.name.lower())
+        switch_data = self.system._service_instance(uri)
+        self.raw = switch_data
+        return self.raw
+
+    @property
+    def name(self):
+        return self.raw.get("name", None)
+
+    @property
+    def uuid(self):
+        return self.raw.get("uuid", None)
+
+    @property
+    def ipv4_assignments(self):
+        ip_interfaces = self.raw.get('ipInterfaces', None)
+        ipv4_assignments = []
+        if ip_interfaces:
+            for interface in ip_interfaces:
+                if "IPv4assignments" in interface:
+                    ipv4_assignments.append(interface['IPv4assignments'])
+        return ipv4_assignments
+
+    @property
+    def ipv6_assignments(self):
+        ip_interfaces = self.raw.get('ipInterfaces', None)
+        ipv6_assignments = []
+        if ip_interfaces:
+            for interface in ip_interfaces:
+                if "IPv6assignments" in interface:
+                    ipv6_assignments.append(interface['IPv6assignments'])
+        return ipv6_assignments
+
+    @property
+    def ports(self):
+        ''' Filters ports data that are also filtered by the provider's parser '''
+        raw_ports = self.raw.get('ports', None)
+        ports = []
+        if raw_ports:
+            for port in raw_ports:
+                port_to_append = {'portName': port.get('portName', None),
+                                'portType': port.get('port', None),
+                                'vlanEnabled': 'PVID' in port,
+                                'peerMacAddress': port.get('peerMacAddress', None)}
+                ports.append(port_to_append)
+        return ports
+
+    def delete(self):
+        data = {"endpoints": [{
+                "ipAdresses": self.ipv4_addresses,
+                "type": self.type,
+                "uuid": self.uuid}],
+            "forceUnmanage": True}
+        self.system._service_post(path="unmanageRequest", request=data)
+
+    def cleanup(self):
+        self.delete()
+
+    def __eq__(self, other):
+        return self.uuid == other.uuid
```

Comment:
Why implement `__eq__` directly when you have already defined `_identifying_attrs` property, which is called by `wrapanapi.Entity.__eq__`?

Reply:
yeah, didn't see that. good point.